src/Object/BaseApiResource.php

```diff
@@ -66,4 +60,40 @@ public function exists()
     {
         return isset($this->data[static::FIELD_ID]);
     }
+
+    /**
+     * @param $data
+     */
+    public function setDataWithoutTracking($data)
+    {
+        $this->data = $data;
+    }
+
+    /**
+     * @return BaseApiResource
+     */
+    public function createFreshCopy()
+    {
+
+        $copy = clone $this;
+        unset($copy->id);
+        return $copy;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getMutableData()
+    {
+        return array_diff_key($this->getData(), array_flip($this->getReadOnlyFieldNames()));
```

Comment:
Fairly sure I've seen this exact code multiple times before in different resources?